src/test/java/io/r2dbc/h2/codecs/BlobCodecTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.h2.codecs;
+
+import io.r2dbc.spi.Blob;
+import org.h2.value.Value;
+import org.h2.value.ValueLobDb;
+import org.h2.value.ValueNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.nio.ByteBuffer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+final class BlobCodecTest {
+
+    ValueLobDb TEST_BLOB;
+
+    @BeforeEach
+    void setUp() {
+        TEST_BLOB = ValueLobDb.createSmallLob(Value.BLOB, new byte[]{1, 2, 3, 4});
+    }
+
+    @Test
+    void decode() {
+        Mono.from(new BlobCodec().decode(TEST_BLOB, Blob.class)
+            .stream())
+            .as(StepVerifier::create)
+            .expectNextMatches(byteBuffer -> {
+                byte[] bytes = byteBuffer.array();
+
+                assertThat(bytes[0]).isEqualTo(Byte.parseByte("1"));
```

Comment:
Why you need `Byte.parseByte(String)` here instead of `(byte) 1` and why you can't compare the whole array at once?